bon-macros/src/builder/builder_gen/mod.rs

```diff
@@ -363,6 +376,16 @@ impl BuilderGenCtx {
             #[must_use = #must_use_message]
             #[doc = #docs]
             #allows
+            #[allow(
+                // We use `__private` prefix for all fields intentionally to hide them
+                clippy::struct_field_names,
+
+                // This lint doesn't emerge until you manually expand the macro. Just
+                // because `bon` developers need to expand the macros a lot it makes
+                // sense to just silence it to avoid some noise. This lint is triggered
+                // by the big PhantomData type generated by the macro
+                clippy::type_complexity
```

Comment:
Allow it on the PhantomData then directly?

Reply:
I'll keep it at struct level. This whole struct is generated and in case some other field starts having "big type complexity" this lint will cover it as well. So it's just for the future